### ISSUE TITLE
Black screen while scrolling the screen.

### ISSUE THREAD

**Issue by nagendraK6** [eneim/toro#245]
What kind of issue is this?
I see black screen couple of seconds when i scroll the screen.

Blank screen while scrolling the 
- [ ] Question. If so, declare clearly your **target** and your current implementation with code formatted in markdown.
Cloned the source, build and deployed on  Nexus 6. Even i can see this problem with emulator code.

- [ ] Bug report. If so, please supply the way to reproduce the issue, including addition information as follow:
  - [ ] Toro version. 
3.4

  - [ ] Android version (can be found in Device info). **Rooted/Custom rom devices are not supported**.
  - [ ] Device name (eg: Nexus 5X, Samsung Galaxy S8)
Nexus 6
  - [ ] Source code that can reproduce the issue (a snapshot is expected).
  - [ ] Logcat stack trace in case of fatal issue.
  
- [ ] Feature request. Start by telling me what problem you’re trying to solve. Don’t send pull requests to implement new features without first getting my support. Sometimes I leave features out on purpose to keep the project small.

- [ ] Other. Senseless issues are normally ignored. Please understand and cooperate.

**Comment by eneim** [comment]
@nagendraK6 It is due to loading behaviour, you can implement a workaround by putting an imageview on top of your VideoView ...

**Comment by nagendraK6** [comment]
Right..that is what i am thinking..but how do we find on state change..like whether player is playing or on pause...and then state change from on buffering to on playing. Any code pointer ?

**Comment by nabeelazam** [comment]
@eneim first I tried setting default art work for shutter view but black screen still appears. 

I am using SimpleExoPlayerView in view holder. I tried using my own image view to handle this issue but unable to catch where I can get callback if the video has started. Can you please elaborate where I can catch that event to handle my own custom image.

**Comment by ghataa** [comment]
@nabeelazam try to add ToroPlayer.EventListener to the ExoPlayerViewHelper class in your ViewHolder:
`static class ViewHolder extends RecyclerView.ViewHolder implements ToroPlayer {

        ...

        private ToroPlayer.EventListener toroPlayerEventListener = new EventListener() {
            @Override
            public void onBuffering() {
                // do something
            }

            @Override
            public void onPlaying() {
                // do something
            }

            @Override
            public void onPaused() {
                // do something
            }

            @Override
            public void onCompleted() {
                // do something
            }
        };

        ...

        @Override
        public void initialize(@NonNull Container container, @Nullable PlaybackInfo playbackInfo) {
            if (mediaUri != null) {

                if (helper == null) {
                    helper = new ExoPlayerViewHelper(this, mediaUri);
                    helper.addPlayerEventListener(toroPlayerEventListener);
                }

                helper.initialize(container, playbackInfo);
            }
        }

        ...

        @Override
        public void release() {
            if (helper != null) {
                helper.removePlayerEventListener(toroPlayerEventListener);
                helper.release();
                helper = null;
            }
        }
}`

**Comment by lucazin** [comment]
Hi @ghataa  i'm trying to set a listener like u said

toroPlayerEventListener = new EventListener() {

but not work..

only work this
`private final Playable.EventListener listener = new Playable.DefaultEventListener()
//        {
//            @Override public void onPlayerStateChanged(boolean playWhenReady, int playbackState)
//            {
//                super.onPlayerStateChanged(playWhenReady, playbackState);
//
//                if(playbackState == PlaybackState.STATE_CONNECTING)
//                    loading_camera.setVisibility(View.VISIBLE);
//
//                if(playbackState == PlaybackState.STATE_BUFFERING)
//                    loading_camera.setVisibility(View.VISIBLE);
//
//                if(playbackState == PlaybackState.STATE_PAUSED)
//                    loading_camera.setVisibility(View.VISIBLE);
//
//                if(playbackState == PlaybackState.STATE_STOPPED)
//                    loading_camera.setVisibility(View.VISIBLE);
//
//                if(playbackState == PlaybackState.STATE_PLAYING)
//                    loading_camera.setVisibility(View.GONE);
//
//
//                player_state.setText(format(Locale.getDefault(), "STATE: %d・PWR: %s", playbackState, playWhenReady));
//            }
//        };`

But black screen still appear =/

**Comment by krhitesh** [comment]
Hey @eneim but onPaused is not called when scrolling with snap helper is enabled. Any ideas?
